lib/widget/triple_state_switch.dart

```diff
@@ -0,0 +1,296 @@
+import 'package:ensemble/framework/event.dart';
+import 'package:ensemble/framework/extensions.dart';
+import 'package:ensemble/screen_controller.dart';
+import 'package:ensemble/util/utils.dart';
+import 'package:ensemble/widget/helpers/widgets.dart';
+import 'package:ensemble/widget/input/form_helper.dart';
+import 'package:ensemble/framework/action.dart' as framework;
+import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
+import 'package:flutter/material.dart';
+
+class EnsembleTripleStateSwitch extends StatefulWidget
+    with
+        Invokable,
+        HasController<TripleStateSwitchController,
+            EnsembleTripleStateSwitchState> {
+  static const type = 'TripleStateSwitch';
+  EnsembleTripleStateSwitch({Key? key}) : super(key: key);
+
+  final TripleStateSwitchController _controller = TripleStateSwitchController();
+  @override
+  TripleStateSwitchController get controller => _controller;
+
+  @override
+  State<StatefulWidget> createState() => EnsembleTripleStateSwitchState();
+
+  @override
+  Map<String, Function> getters() {
+    return {'value': () => _controller.value.name};
+  }
+
+  @override
+  Map<String, Function> methods() {
+    return {};
+  }
+
+  @override
+  Map<String, Function> setters() {
+    return {
+      'value': (value) => _controller.value =
+          SwitchState.values.from(Utils.getString(value, fallback: 'start')) ??
+              SwitchState.start,
+      'leadingText': (text) =>
+          _controller.leadingText = Utils.optionalString(text),
+      'trailingText': (text) =>
+          _controller.trailingText = Utils.optionalString(text),
+      'activeColor': (color) => _controller.activeColor = Utils.getColor(color),
+      'inactiveColor': (color) =>
+          _controller.inactiveColor = Utils.getColor(color),
+      'intermediateColor': (color) =>
+          _controller.intermediateColor = Utils.getColor(color),
+      'activeThumbColor': (color) =>
+          _controller.activeThumbColor = Utils.getColor(color),
+      'inactiveThumbColor': (color) =>
+          _controller.inactiveThumbColor = Utils.getColor(color),
+      'onChange': (definition) => _controller.onChange =
+          framework.EnsembleAction.fromYaml(definition, initiator: this)
+    };
+  }
+
+  void onToggle(SwitchState newValue) {
+    setProperty('value', newValue.name);
+  }
+}
+
+class TripleStateSwitchController extends FormFieldController {
+  SwitchState value = SwitchState.start;
+  String? leadingText;
+  String? trailingText;
+  Color? activeColor;
+  Color? activeThumbColor;
+  Color? inactiveColor;
+  Color? inactiveThumbColor;
+  Color? intermediateColor;
+
+  framework.EnsembleAction? onChange;
+}
+
+class EnsembleTripleStateSwitchState
+    extends FormFieldWidgetState<EnsembleTripleStateSwitch> {
+  void onToggle(SwitchState newValue) {
+    widget.onToggle(newValue);
+    widget._controller.value = newValue;
+    if (widget._controller.onChange != null) {
```

Comment:
add a dispatch event here, so if any listener is bind to switch state change will update as well. 
Something like this 
https://github.com/EnsembleUI/ensemble/blob/afa3e8dcfa0eb0e6fbc16a783f03426b2b77b71f/lib/layout/tab_bar.dart#L160-L161

Reply:
Yes, my bad. It works. For some reason i thought the case of tabbar and footer not updating on tab change because of event but I was wrong it update I check my theory.